syntax = "proto3";

message ApplicationMessage {
  enum MessageType {
      IDENTIFICATION = 0;
      COMMAND = 1;
  }

  MessageType type = 1;
  string command = 2;
  string args = 3;
}

message Object {
  string address = 1;
  string type = 2;
  string status = 3;
  int32 temp = 4;
}

message GatewayMessage {
  enum MessageType {
      UPDATE = 0;
      GET = 1;
      LIST = 2;
  }

  MessageType response_type = 1;
  repeated Object object = 2;
}